drivers/ethernet/Kconfig.e1000

```diff
@@ -0,0 +1,21 @@
+# Kconfig - Intel(R) PRO/1000 Gigabit Ethernet driver configuration options
+
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig ETH_E1000
+	bool
+	prompt "Intel(R) PRO/1000 Gigabit Ethernet driver"
+	depends on NET_L2_ETHERNET PCI_ENUMERATION
+	help
+	  Enable Intel(R) PRO/1000 Gigabit Ethernet driver.
+
+config ETH_E1000_IRQ
```

Comment:
Add dt support, and get this property from there.

Reply:
E1000 is a PCI only device and everything is discoverable through a PCI scan (including the IRQ number).

However, due to limitations of the IRQ_CONNECT() macro, where all parameters should be compile time constants, the IRQ line presently has to be configurable though Kconfig.

I briefly looked into device tree and can’t see any PCI or Ethernet support or similar use cases.

Can you be more specific with what kind of node, behind what bus, etc, should be defined in the device tree?

At the end, this still will need to be configurable (we would like to support a few other emulators in addition to qemu), unless the limitations of IRQ_CONNECT() are removed.